doc/source/math/resizing_ops.rst

```diff
@@ -0,0 +1,319 @@
+
+with the zero'th order moments
+
+.. math::
+    M_{\mathrm{l},0}(y) := \sum_{j=0}^{p-1} y_j, \quad M_{\mathrm{r},0}(y) := \sum_{j=p+n}^{m-1} y_j.
+
+Hence, we get
+
+.. math::
+    P_{\mathrm{o0}}^*(y)_i :=
+    \begin{cases}
+    M_{\mathrm{l},0}(y), & \text{if } i = 0,     \\
+    M_{\mathrm{r},0}(y), & \text{if } i = n - 1, \\
+    0,                   & \text{else},
+    \end{cases}
+
+with the convention that the sum of the two values is taken in the case that $n = 1$, i.e. both first cases are the same.
+Hence, after constructing the restriction :math:`x \in \mathbb{R}^n` of a vector :math:`y \in \mathbb{R}^m` to the main part :math:`p, \dots, p + n - 1`, the sum of the entries to the left are added to :math:`x_0`, and the sum of the entries to the right are added to :math:`x_{n-1}`.
+
+
+Order 1 padding
+---------------
+In this padding mode, a given vector is continued with constant slope instead of constant value, i.e.
+
+.. math::
+ E_{\mathrm{o1}}(x)_j :=
+ \begin{cases}
+  x_0 + (j - p)(x_1 - x_0),                     & \text{if } 0 \leq j \leq p - 1,      \\
+  x_{j-p},                                      & \text{if } p \leq j \leq p + n - 1,  \\
+  x_{n-1} + (j - p - n + 1)(x_{n-1} - x_{n-2}), & \text{if } p + n \leq j \leq m - 1.
+ \end{cases}
+ :label: def_order1_pad_op
+
+We can write this operator as :math:`E_{\mathrm{o1}} = E_{\mathrm{o0}} + S_{\mathrm{o1}}` with the order-1 specific part
+
+.. math::
+    S_{\mathrm{o1}}(x)_j :=
+    \begin{cases}
+    (j - p)(x_1 - x_0),                 & \text{if } 0 \leq j \leq p - 1,      \\
+    0,                                  & \text{if } p \leq j \leq p + n - 1,  \\
+    (j - p - n + 1)(x_{n-1} - x_{n-2}), & \text{if } p + n \leq j \leq m - 1.
+    \end{cases}
+
+For its adjoint, we get
+
+.. math::
+ \langle S_{\mathrm{o1}}(x), y \rangle_{\mathbb{R}^m}
+ &= \sum_{j=0}^{p-1} (j - p)(x_1 - x_0)\, y_j +
+    \sum_{j=p+n}^{m-1} (j - p - n + 1)(x_{n-1} - x_{n-2})\, y_j \\
+ &= x_0 (-M_{\mathrm{l}}(y)) + x_1 M_{\mathrm{l}}(y) +
+    x_{n-2}(-M_{\mathrm{r}}(y)) + x_{n-1} M_{\mathrm{r}}(y)
+
+with the first order moments
+
+.. math::
+ M_{\mathrm{l},1}(y) := \sum_{j=0}^{p-1} (j - p)\, y_j, \quad
+ M_{\mathrm{r},1}(y) := \sum_{j=p+n}^{m-1} (j - p - n + 1)\, y_j.
+
+Hence, the order-1 specific operator has the adjoint
+
+.. math::
+    S_{\mathrm{o1}}^*(y)_i :=
+    \begin{cases}
+    -M_{\mathrm{l},1}(y), & \text{if } i = 0,     \\
+    M_{\mathrm{l},1}(y),  & \text{if } i = 1,     \\
+    -M_{\mathrm{r},1}(y), & \text{if } i = n - 2, \\
+    M_{\mathrm{r},1}(y),  & \text{if } i = n - 1, \\
+    0,                  & \text{else},
+    \end{cases}
+
+with the convention of summing values for overlapping cases, i.e. if :math:`i \in \{1, 2\}`.
+In practice, the adjoint for the order 1 padding case is applied by computing the zero'th and first order moments of :math:`y` and adding them to the two outmost entries of :math:`x` according to the above rule.
```

Comment:
It's like unfolding (extension, forward) and folding (restriction, adjoint) a piece of paper. The contributions are summed. I don't know how that plays along with interpolation schemes, but they're not really involved here.